nautobot/dcim/migrations/0020_alter_device_and_rack_role_add_new_role.py

```diff
@@ -0,0 +1,62 @@
+# Generated by Django 3.2.16 on 2022-11-25 11:45
+
+from django.db import migrations, models
+import django.db.models.deletion
+import nautobot.extras.models.roles
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("extras", "0055_collect_roles_from_related_apps_roles"),
+        ("dcim", "0019_device_redundancy_group_data_migration"),
+    ]
+
+    operations = [
+        # ##########
+        # Device
+        # ##########
+        migrations.AlterField(
+            model_name="device",
+            name="device_role",
+            field=models.ForeignKey(
+                on_delete=django.db.models.deletion.PROTECT,
+                related_name="devices",
+                to="dcim.devicerole",
+                null=True,
+            ),
+        ),
```

Comment:
What's changing here? I notice no corresponding AlterField on the Rack `role` field below.

Reply:
I had to set `device_role` to allow `null` cause when reversing migration i get this error "legacy_role cannot have null values"